tf2_eigen/include/tf2_eigen/tf2_eigen.h

```diff
@@ -314,6 +314,12 @@ geometry_msgs::Pose toMsg(const Eigen::Affine3d& in) {
   msg.orientation.y = q.y();
   msg.orientation.z = q.z();
   msg.orientation.w = q.w();
+  if (msg.orientation.w < 0) {
```

Comment:
Constrain quaternions to the half-space w >= 0 as in [`tf1`](https://github.com/ros/geometry/blob/bea847cf721894517d2da88b26052c9a1d427251/eigen_conversions/src/eigen_msg.cpp#L104-L109).